webui/src/Buttons/EditButton.tsx

```diff
@@ -372,6 +373,59 @@ interface TabsSectionProps {
 	feedbacks: FeedbackInstance[]
 }
 
+interface TabStepProps{
+	selectedKey: string | false
+	k: string | number
+	i: number
+	name:string | undefined
+	inputState: {showInputEle:boolean}
+	linkClassname: string | undefined
+	socket: CompanionSocketType
+	controlId: string
+
+}
+function TabStep({selectedKey, k, i, name, inputState, linkClassname, socket, controlId }: TabStepProps){
+
+	const renameStep = useCallback(
+		(stepId: string, newName: string) => {
+			socketEmitPromise(socket, 'controls:step:rename', [controlId, stepId, newName]).catch((e) => {
+				console.error('Failed to rename step:', e)
+			})
+		},
+		[socket, controlId]
+	)
+	
+	function onChangeHandler(e: React.ChangeEvent<HTMLInputElement>){
+		setState({showInputEle: state.showInputEle});
+		if(selectedKey){
+			renameStep(selectedKey, e.target.value);
+		}
+	}
+
+	const [state, setState] = useState(inputState);
+	return (
+		<CNavItem key={k} className="nav-steps-special">
+			{state.showInputEle? (
+				<CNavLink className={linkClassname}>
+				<input
+					type="text"
+					value={name}
+					onChange={(e)=>{onChangeHandler(e)}}
+					onKeyDown={(e)=>{e.key === 'Enter' && setState({showInputEle: false})}}
+					onBlur={(e)=>setState({showInputEle: false})}
+					autoFocus
+				>			
+				</input>
+				</CNavLink>
+			) : (
+				<CNavLink onDoubleClick={(e)=>setState({showInputEle: true})} data-tab={`step:${k}`} className={linkClassname}>
```

Comment:
This is something for me to think on rather than something for you to resolve 

We don't currently do double click to edit anywhere, It might be good to have some visual hint to help users discover this. Perhaps a `title` would be enough?
Double click to edit could also make sense elsewhere too.

Or discoverability could be solved by showing an edit pencil for just the current step? Would that look too busy?